--- mujs-1.0.7/Makefile.orig	2020-03-17 14:10:18.000000000 +0100
+++ mujs-1.0.7/Makefile	2020-05-10 15:47:59.561908525 +0200
@@ -16,6 +16,8 @@
 # Compiler flags for various configurations:
 
 CFLAGS := -std=c99 -pedantic -Wall -Wextra -Wno-unused-parameter
+LTCC = libtool --mode=compile --tag=CC $(CC)
+LTLINK = libtool --mode=link --tag=CC $(CC)
 
 ifeq "$(CC)" "clang"
   CFLAGS += -Wunreachable-code
@@ -52,7 +54,7 @@
 default: shell
 shell: $(OUT)/mujs $(OUT)/mujs-pp
 static: $(OUT)/libmujs.a
-shared: $(OUT)/libmujs.so
+shared: $(OUT)/libmujs.la
 
 astnames.h: jsparse.h
 	grep -E '(AST|EXP|STM)_' jsparse.h | sed 's/^[^A-Z]*\(AST_\)*/"/;s/,.*/",/' | tr A-Z a-z > $@
@@ -77,13 +79,14 @@
 	@ mkdir -p $(dir $@)
 	$(AR) cr $@ $^
 
-$(OUT)/libmujs.so: one.c $(HDRS)
-	@ mkdir -p $(dir $@)
-	$(CC) $(CFLAGS) -fPIC -shared -o $@ $< -lm
+$(OUT)/%.lo: one.c $(HDRS)
+	$(LTCC) $(CFLAGS) -o $@ -c $<
 
-$(OUT)/mujs: $(OUT)/libmujs.o $(OUT)/main.o
-	@ mkdir -p $(dir $@)
-	$(CC) $(LDFLAGS) -o $@ $^ $(LIBREADLINE) -lm
+$(OUT)/libmujs.la: $(OUT)/libmujs.o
+	$(LTLINK) $(LDFLAGS) -o $@ $^ -rpath $(libdir) -lm
+
+$(OUT)/mujs: $(OUT)/libmujs.la $(OUT)/main.o
+	$(LTLINK) $(LDFLAGS) -o $@ $^ $(LIBREADLINE) -lm
 
 $(OUT)/mujs-pp: $(OUT)/libmujs.o $(OUT)/pp.o
 	@ mkdir -p $(dir $@)
@@ -115,9 +118,10 @@
 	install -m 644 $(OUT)/libmujs.a $(DESTDIR)$(libdir)
 
-install-shared: install-common $(OUT)/libmujs.so
-	install -m 755 $(OUT)/libmujs.so $(DESTDIR)$(libdir)
+install-shared: install-common $(OUT)/libmujs.la
+	libtool --mode=install install $(OUT)/libmujs.la $(DESTDIR)$(libdir)
+	libtool --mode=install install $(OUT)/mujs $(DESTDIR)$(bindir)
 
-install: install-static
+install: install-static install-shared
 
 uninstall:
 	rm -f $(DESTDIR)$(bindir)/mujs
