--- mujs/Makefile.orig	2016-04-14 21:02:05.000000000 +0200
+++ mujs/Makefile	2016-04-14 21:20:51.139270570 +0200
@@ -16,6 +16,8 @@
 # Compiler flags for various configurations:
 
 CFLAGS := -std=c99 -pedantic -Wall -Wextra -Wno-unused-parameter
+LTCC = libtool --mode=compile --tag=CC $(CC)
+LTLINK = libtool --mode=link --tag=CC $(CC)
 
 ifeq "$(CC)" "clang"
   CFLAGS += -Wunreachable-code
@@ -53,7 +55,7 @@
 default: shell
 shell: $(OUT)/mujs $(OUT)/mujs-pp
 static: $(OUT)/libmujs.a
-shared: $(OUT)/libmujs.so
+shared: $(OUT)/libmujs.la
 
 astnames.h: jsparse.h
 	grep -E '(AST|EXP|STM)_' jsparse.h | sed 's/^[^A-Z]*\(AST_\)*/"/;s/,.*/",/' | tr A-Z a-z > $@
@@ -78,13 +80,14 @@
 	@ mkdir -p $(dir $@)
 	$(AR) cr $@ $^
 
-$(OUT)/libmujs.so: one.c $(HDRS)
-	@ mkdir -p $(dir $@)
-	$(CC) $(CFLAGS) -fPIC -shared -o $@ $< -lm
-
-$(OUT)/mujs: $(OUT)/libmujs.o $(OUT)/main.o
-	@ mkdir -p $(dir $@)
-	$(CC) $(LDFLAGS) -o $@ $^ $(LIBREADLINE) -lm
+$(OUT)/%.lo: one.c $(HDRS)
+	$(LTCC) $(CFLAGS) -o $@ -c $<
+
+$(OUT)/libmujs.la: $(OUT)/libmujs.o
+	$(LTLINK) $(LDFLAGS) -o $@ $^ -rpath $(libdir) -lm
+
+$(OUT)/mujs: $(OUT)/libmujs.la $(OUT)/main.o
+	$(LTLINK) $(LDFLAGS) -o $@ $^ $(LIBREADLINE) -lm
 
 $(OUT)/mujs-pp: $(OUT)/libmujs.o $(OUT)/pp.o
 	@ mkdir -p $(dir $@)
@@ -118,9 +124,10 @@
 	install -m 644 $(OUT)/libmujs.a $(DESTDIR)$(libdir)
 
 install-shared: install-common shared
-	install -m 755 $(OUT)/libmujs.so $(DESTDIR)$(libdir)
+	libtool --mode=install install $(OUT)/libmujs.la $(DESTDIR)$(libdir)
+	libtool --mode=install install $(OUT)/mujs $(DESTDIR)$(bindir)
 
-install: install-static
+install: install-static install-shared
 
 tarball:
 	git archive --format=zip --prefix=mujs-$(VERSION)/ HEAD > mujs-$(VERSION).zip
