--- mujs-1.2.0/Makefile.orig	2021-12-08 12:56:12.000000000 +0100
+++ mujs-1.2.0/Makefile	2022-04-08 06:21:16.629112683 +0200
@@ -88,13 +88,13 @@ $(OUT)/libmujs.a: $(OUT)/libmujs.o
 
 $(OUT)/libmujs.$(SO_EXT): one.c $(HDRS)
 	@ mkdir -p $(dir $@)
-	$(CC) $(CFLAGS) $(CPPFLAGS) -fPIC -shared $(LDFLAGS) -o $@ $< -lm
+	$(CC) $(CFLAGS) $(CPPFLAGS) -fPIC -shared $(LDFLAGS) -Wl,-soname,libmujs.so.0 -o $@ $< -lm
 
-$(OUT)/mujs: $(OUT)/libmujs.o $(OUT)/main.o
+$(OUT)/mujs: $(OUT)/main.o $(OUT)/libmujs.$(SO_EXT)
 	@ mkdir -p $(dir $@)
 	$(CC) $(LDFLAGS) -o $@ $^ $(LIBREADLINE) -lm
 
-$(OUT)/mujs-pp: $(OUT)/libmujs.o $(OUT)/pp.o
+$(OUT)/mujs-pp: $(OUT)/pp.o $(OUT)/libmujs.$(SO_EXT)
 	@ mkdir -p $(dir $@)
 	$(CC) $(LDFLAGS) -o $@ $^ -lm
 
@@ -124,7 +124,9 @@ install-static: install-common $(OUT)/li
 	install -m 644 $(OUT)/libmujs.a $(DESTDIR)$(libdir)
 
 install-shared: install-common $(OUT)/libmujs.$(SO_EXT)
-	install -m 755 $(OUT)/libmujs.$(SO_EXT) $(DESTDIR)$(libdir)
+	install -m 755 $(OUT)/libmujs.$(SO_EXT) $(DESTDIR)$(libdir)/libmujs.so.0.0.0
+	ln -s libmujs.so.0.0.0 $(DESTDIR)$(libdir)/libmujs.so.0
+	ln -s libmujs.so.0.0.0 $(DESTDIR)$(libdir)/libmujs.so
 
 install: install-static
 
