--- mujs-1.3.3/Makefile.orig	2023-04-15 12:09:09.276756511 +0200
+++ mujs-1.3.3/Makefile	2023-04-15 12:22:41.608077963 +0200
@@ -92,16 +92,16 @@
 
 build/release/libmujs.$(SO): one.c $(SRCS) $(HDRS)
 	@mkdir -p $(@D)
-	$(CC) $(CFLAGS) $(OPTIM) -fPIC -shared -o $@ one.c -lm
+	$(CC) $(CFLAGS) $(OPTIM) -fPIC -shared $(LDFLAGS) -Wl,-soname,libmujs.so.0 -o $@ one.c -lm
 build/release/libmujs.o: one.c $(SRCS) $(HDRS)
 	@mkdir -p $(@D)
 	$(CC) $(CFLAGS) $(OPTIM) -c -o $@ one.c
 build/release/libmujs.a: build/release/libmujs.o
 	$(AR) cr $@ $^
-build/release/mujs: main.c build/release/libmujs.o
-	$(CC) $(CFLAGS) $(OPTIM) -o $@ $^ -lm -DHAVE_READLINE -lreadline
-build/release/mujs-pp: pp.c build/release/libmujs.o
-	$(CC) $(CFLAGS) $(OPTIM) -o $@ $^ -lm
+build/release/mujs: main.c build/release/libmujs.$(SO)
+	$(CC) $(CFLAGS) $(OPTIM) $(LDFLAGS) -o $@ $^ -lm -DHAVE_READLINE -lreadline
+build/release/mujs-pp: pp.c build/release/libmujs.$(SO)
+	$(CC) $(CFLAGS) $(OPTIM) $(LDFLAGS) -o $@ $^ -lm
 
 build/release/mujs.pc:
 	@mkdir -p $(@D)
@@ -126,7 +126,9 @@
 	install -m 644 build/release/libmujs.a $(DESTDIR)$(libdir)
 
 install-shared: install-common build/release/libmujs.$(SO)
-	install -m 755 build/release/libmujs.$(SO) $(DESTDIR)$(libdir)
+	install -m 755 build/release/libmujs.$(SO) $(DESTDIR)$(libdir)/libmujs.so.0.0.0
+	ln -s libmujs.so.0.0.0 $(DESTDIR)$(libdir)/libmujs.so.0
+	ln -s libmujs.so.0.0.0 $(DESTDIR)$(libdir)/libmujs.so
 
 install: install-static
 
