--- mujs-1.1.3/Makefile.orig	2021-06-08 13:41:51.000000000 +0200
+++ mujs-1.1.3/Makefile	2021-07-16 21:24:29.059361193 +0200
@@ -18,6 +18,8 @@ endif
 # Compiler flags for various configurations:
 
 CFLAGS := -std=c99 -pedantic -Wall -Wextra -Wno-unused-parameter
+LTCC = libtool --mode=compile --tag=CC $(CC)
+LTLINK = libtool --mode=link --tag=CC $(CC)
 
 ifeq "$(CC)" "clang"
   CFLAGS += -Wunreachable-code
@@ -55,7 +57,7 @@ HDRS := $(wildcard js*.h mujs.h utf.h re
 default: shell
 shell: $(OUT)/mujs $(OUT)/mujs-pp
 static: $(OUT)/libmujs.a
-shared: $(OUT)/libmujs.so
+shared: $(OUT)/libmujs.la
 
 astnames.h: jsparse.h
 	grep -E '(AST|EXP|STM)_' jsparse.h | sed 's/^[^A-Z]*\(AST_\)*/"/;s/,.*/",/' | tr A-Z a-z > $@
@@ -80,13 +82,15 @@ $(OUT)/libmujs.a: $(OUT)/libmujs.o
 	@ mkdir -p $(dir $@)
 	$(AR) cr $@ $^
 
-$(OUT)/libmujs.so: one.c $(HDRS)
+$(OUT)/libmujs.lo: one.c $(HDRS)
 	@ mkdir -p $(dir $@)
-	$(CC) $(CFLAGS) $(CPPFLAGS) -fPIC -shared $(LDFLAGS) -o $@ $< -lm
+	$(LTCC) $(CFLAGS) $(CPPFLAGS) -o $@ -c $<
 
-$(OUT)/mujs: $(OUT)/libmujs.o $(OUT)/main.o
-	@ mkdir -p $(dir $@)
-	$(CC) $(LDFLAGS) -o $@ $^ $(LIBREADLINE) -lm
+$(OUT)/libmujs.la: $(OUT)/libmujs.lo
+	$(LTLINK) $(LDFLAGS) -o $@ $^ -rpath $(libdir) -lm
+
+$(OUT)/mujs: $(OUT)/libmujs.la $(OUT)/main.o
+	$(LTLINK) $(LDFLAGS) -o $@ $^ $(LIBREADLINE) -lm
 
 $(OUT)/mujs-pp: $(OUT)/libmujs.o $(OUT)/pp.o
 	@ mkdir -p $(dir $@)
@@ -117,10 +121,11 @@ install-common: $(OUT)/mujs $(OUT)/mujs.
 install-static: install-common $(OUT)/libmujs.a
 	install -m 644 $(OUT)/libmujs.a $(DESTDIR)$(libdir)
 
-install-shared: install-common $(OUT)/libmujs.so
-	install -m 755 $(OUT)/libmujs.so $(DESTDIR)$(libdir)
+install-shared: install-common $(OUT)/libmujs.la
+	libtool --mode=install install $(OUT)/libmujs.la $(DESTDIR)$(libdir)
+	libtool --mode=install install $(OUT)/mujs $(DESTDIR)$(bindir)
 
-install: install-static
+install: install-shared
 
 uninstall:
 	rm -f $(DESTDIR)$(bindir)/mujs
